score/parse.go

```diff
@@ -0,0 +1,40 @@
+package score
+
+import (
+	"encoding/json"
+	"errors"
+	"strings"
+)
+
+// hiddenSecret is used to replace the global secret when parsing.
+var hiddenSecret = "hidden"
+
+// ErrScoreNotFound is returned if the parse string did not contain a
+// JSON score string.
+var ErrScoreNotFound = errors.New("score not found in string")
+
+// Parse returns a score object for the provided JSON string s
+// which contains secret.
+func Parse(s, secret string) (*Score, error) {
+	if strings.Contains(s, secret) {
+		var sc Score
+		err := json.Unmarshal([]byte(s), &sc)
+		if err == nil {
+			if sc.Secret == secret {
+				sc.Secret = hiddenSecret // overwrite secret
+			}
+			return &sc, nil
+		}
+		if strings.Contains(err.Error(), secret) {
+			// this is probably not necessary, but to be safe
+			return nil, errors.New("error suppressed to avoid revealing secret")
+		}
+		return nil, err
+	}
+	return nil, ErrScoreNotFound
+}
+
+// HasPrefix returns true if the provided string s has a parsable prefix string.
```

Comment:
Currently it is being used in `BuildResult.Add()` in the `ci-refactor` branch. 

I think it makes sense to limit how test cases can interact with AG. Otherwise, we can easily get ourselves into trouble with complicated pattern matching that can lead to bugs. Lets keep things simple.

I don't think that JSON writer libraries will violate the order given by a data structure. If this is the case, and it can't easily be fixed elsewhere, we can revisit this. Do you agree?